tools.go

```diff
@@ -0,0 +1,12 @@
+// +build tools
```

Comment:
What would the relationship be between this file and go.mod? Is this adding versioned dependencies that would be pruned by `go mod tidy` when not making a `tools` build?

Could `tools.go` live in the `protoc/` directory with an isolated go.mod in that directory? (`go list -m all` could be used to list the dependencies if that were the case.)

We run into the problem that users will have the dependencies already installed, but at the wrong versions of those dependencies, resulting in proto file jitter depending on the committer.

The biggest potential benefit of this change is that the protobuf tooling version used to create the generated files would match the ones that are used in other parts of this project.  If that is the goal, then the containerized protoc.sh approach should be removed.

The proto builds must be repeatable.  I would like to see a protoc build added to the Github actions to look for proto file diffs that were not incorporated in a commit.

If controlling the local version of the proto tools is problematic, I would recommend that we use our own Dockerfile to control the tooling used in the proto file builds.

Reply:
> What would the relationship be between this file and go.mod? Is this adding versioned dependencies that would be pruned by `go mod tidy` when not making a `tools` build?

This file is used in harmony with go.mod. This is coming from one of the options in the Go Wiki, golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module.

> Could `tools.go` live in the `protoc/` directory with an isolated go.mod in that directory? (`go list -m all` could be used to list the dependencies if that were the case.)

I'd prefer not to do that. For one it seems weird to define a mod in a sub directory since mods are meant to be at a repo root right? Having these deps in the top-level mod also ensures they are always being used.
 
> We run into the problem that users will have the dependencies already installed, but at the wrong versions of those dependencies, resulting in proto file jitter depending on the committer.

This takes care of that since we'd be using the pinned versions of the tools from the go.mod file.
 
> The biggest potential benefit of this change is that the protobuf tooling version used to create the generated files would match the ones that are used in other parts of this project. If that is the goal, then the containerized protoc.sh approach should be removed.

I meant to do this, but forgot. This PR still doesn't deal with `protoc` versions/installations. I deal with that with `shell.nix` personally, the build-in-docker approach in protoc.sh also helps there too. 

> 
> The proto builds must be repeatable. I would like to see a protoc build added to the Github actions to look for proto file diffs that were not incorporated in a commit.

Can do.

> If controlling the local version of the proto tools is problematic, I would recommend that we use our own Dockerfile to control the tooling used in the proto file builds.

I don't particularly like leaning so much on docker for all dev tooling, though I think the happy medium is a nix docker image that reads the shell.nix file and uses those versions :D.